src/api/analytics/gtag.js

```diff
@@ -0,0 +1,192 @@
+  window.gtag('event', eventName, {
+    ...value,
+    non_interaction: nonInteraction
+  })
+}
+
+/* --- General --- */
+const emailSubscription = (from) =>
+  event('email_subscription', { subscription_click_from: from })
+
+/* --- Compendia --- */
+const compendiaDownload = (type, organism) =>
+  event(`compendia_${type}_download`, {
+    [`${type}_organism`]: organism
+  })
+
+/* --- Datasets --- */
+const myDatasetAction = (action) =>
+  event('my_dataset_action', {
+    my_dataset_action: `${action[0].toUpperCase()}${action.substring(
+      1
+    )} Samples`
+  })
+
+const myDatasetDownloadOptions = (dataset) =>
+  event('my_dataset_download_options', {
+    my_dataset_download_options: `Aggregate: ${
+      dataset.aggregate_by
+    }, Transformation: ${transformation[dataset.scale_by]}, QN: ${
+      dataset.quantile_normalize ? 'Skipped' : 'Not skipped'
+    }
+  `
+  })
+
+const datasetDownload = (token) => event('dataset_downalod', { token })
+
+const oneOffExperimentDownload = (accessionCode) =>
+  event('one_off_experiment_download', {
+    one_off_experiment_download: accessionCode
+  })
+
+const regeneratedDataset = (state, defaultOptions, newOptions) => {
+  // due to GA char limit, keys names are abbreviated
+  const format = (option) => {
+    return `A: ${option.aggregate_by}, T: ${
+      transformation[option.scale_by]
+    }, QN: ${option.quantile_normalize ? 'Not skipped' : 'Skipped'}`
+  }
+
+  event('regenerated_dataset', {
+    regenerated_state: state ? 'Expired' : 'Valid',
+    regenerated_download_options: `${format(defaultOptions)} ${
+      newOptions ? `(new) ${format(newOptions)}` : ''
+    }`
+  })
+}
+
+const sharedDataset = (status) =>
+  event('shared_dataset', {
+    shared_state: status ? 'Processed' : 'Unprocessed'
+  })
+
+/* --- Links --- */
+const experimentPageClick = (from) =>
+  event('page_view', {
+    experiment_page_click_from: from
+  })
+
+const exploredUsageClick = (usage, type = 'dataset') =>
+  event(`click`, { [`${type}_explored_usage`]: usage })
+
+const navClick = (item, type = 'global') =>
+  event(`page_view`, {
+    [`${type}_nav_item`]: item
+  })
+
+const outboundClick = (url) => {
+  let parameter
+  switch (true) {
+    case url === links.alsf || url === links.ccdl:
+      parameter = 'alsf_domain_name'
+      break
+    case url === links.ccdl_donate:
+      parameter = 'donate_click_from'
+      break
+    case url.includes(links.refinebio_githubio):
+      parameter = 'alsf_githubio_click_from'
+      break
+    case url.includes(links.alsf_github):
+      parameter = 'alsf_repo_click_from'
+      break
+    case url.includes(links.refinebio_api_docs):
+      parameter = 'api_docs_click_from'
+      break
+    case url.includes(links.refinebio_docs):
+      parameter = 'docs_click_from'
+      break
+    case url.includes(links.ccdl_twitter):
+      parameter = 'social_media'
+      break
+    default:
+      break
+  }
+
+  event('click', {
+    [parameter]: url
+  })
+}
```

Comment:
```suggestion
const outboundClick = (url) => {
  const [linkName] = Object.entries(links).find(([key, value]) => value === url)
  const parameter = `click_to_${linkName}`

  event('click', {
    [parameter]: url
  })
}
```

Can we instead do something like this?

Reply:
Thank you for the great suggestion! 👍  I’ve incorporated your advice with some tweaks and revised `outboundClick` (at 57de1f46b6c47d03a952742fa8a666387967bbf2).

I've added `specialCases` tweak primarily to avoid having long event parameter names (e.g., `click_to_refinebio_github_differential_expression_microarray` that is `60` chars). User-defined event parameters need to be registered as custom definitions in the GA property (e.g., `CCDL > Refine.bio Web - Dev`) to make sure they can be tracked and available in GA reports. Since the event parameter name limit is currently `40` chars [[GA4] Event collection limits](https://support.google.com/analytics/answer/9267744?hl=en), this tweak helps us keep the parameter name within that limit.

Please let me know your thoughts on this!

Based on this change, I’ve also updated the parameter names in the google sheet [Custom Events List > Outbound Link](https://docs.google.com/spreadsheets/d/1qKeq8_pEezvq-OAp6ohCjJhrr9EQYLLyGYFAQe9XvS4/edit?gid=1911374202#gid=1911374202) accordingly, thank you David!